packages/truffle/test/scenarios/migrations/deferred-chain.js

```diff
@@ -0,0 +1,50 @@
+const MemoryLogger = require("../MemoryLogger");
+const CommandRunner = require("../commandRunner");
+const path = require("path");
+const assert = require("assert");
+const Server = require("../server");
+const Reporter = require("../reporter");
+const sandbox = require("../sandbox");
+
+describe("migrate (empty)", function () {
+  let config;
+  const project = path.join(
+    __dirname,
+    "../../sources/migrations/deferred-chain"
+  );
+  const logger = new MemoryLogger();
+
+  before(async function () {
+    await Server.start();
+  });
+  after(async function () {
+    await Server.stop();
+  });
+
+  before(async function () {
+    this.timeout(10000);
+    config = await sandbox.create(project);
+    config.network = "development";
+    config.logger = logger;
+    config.mocha = {
+      reporter: new Reporter(logger)
+    };
+  });
+
+  it("Correctly handles control flow in deployments", async function () {
+    this.timeout(70000);
+
+    try {
+      //the migration fails (due to me being unable to fully fix the
+      //problem), so we have to put it in a try
```

Comment:
Well, there's a rejection, sure, but it's caught.  Caught rejections shouldn't cause an overall rejection.  But because of how the deferred chain currently works, they do.  And I don't presently know how to fix that (and I figure that can perhaps be a separate PR, if we want to put in the effort to fix that before we throw all this out with Truffle 6).

Reply:
(I should also go file an issue for this!)

Reply:
Oh, I see. If the issue was created, can you replace the mea cupla with a link to the issue?

Reply:
Oh, sure, I can do that before merging!

Reply:
(OK, added!)